libglusterfs/src/iobuf.c

```diff
@@ -109,23 +98,17 @@ __iobuf_arena_destroy_iobufs(struct iobuf_arena *iobuf_arena)
         list_del_init(&iobuf->list);
         iobuf++;
     }
-
-    GF_FREE(iobuf_arena->iobufs);
 }
 
 static void
 __iobuf_arena_destroy(struct iobuf_arena *iobuf_arena)
 {
-    GF_VALIDATE_OR_GOTO("iobuf", iobuf_arena, out);
-
-    __iobuf_arena_destroy_iobufs(iobuf_arena);
-
     if (iobuf_arena->mem_base && iobuf_arena->mem_base != MAP_FAILED)
```

Comment:
This check is not really needed. We are never calling `__iobuf_arena_destroy()` without an initialized `mem_base`.

Reply:
Removed.